modules/modes/sweet_scent.py

```diff
@@ -0,0 +1,40 @@
+from asyncio import log
+from typing import Generator
+
+from modules.context import context
+from modules.menu_parsers import CursorOptionEmerald, CursorOptionFRLG, CursorOptionRS
+from modules.menuing import PokemonPartyMenuNavigator, StartMenuNavigator
+from modules.player import get_player_avatar
+from modules.pokemon import get_move_by_name, get_party
+from ._interface import BotMode
+
+
+class SweetScentMode(BotMode):
+    @staticmethod
+    def name() -> str:
+        return "Sweet Scent"
+
+    @staticmethod
+    def is_selectable() -> bool:
+        return get_player_avatar().map_location.has_encounters
+
+    def run(self) -> Generator:
```

Comment:
In this function, you'll want to perform a check that a pokemon has the move `Sweet Scent`. An example of this can be found in the [Rock Smash Mode](https://github.com/40Cakes/pokebot-gen3/blob/34cf5c59527b19a6f22ecd5f8665d7f51d535766/modules/modes/rock_smash.py#L75)

Currently if you enable the mode without a sweet scent pokemon, the bot will return an error:
```
TypeError: '>' not supported between instances of 'int' and 'NoneType'
Shutting down...
```